### PR TITLE
Added hotfix-style changes back to main repository + added some tests for them.

### PR DESCRIPTION
**Pull request by e-kolpakov**

### CODE REVIEW

**Comment by antoviaque** [comment]
@e-kolpakov Thanks! Note that you seem to have a merge conflict on this PR though ("*We can’t automatically merge this pull request.*")

**Comment by e-kolpakov** [comment]
@antoviaque rebased

File: jquery.xblock.js [hunk 1]
```diff
@@ -154,6 +154,15 @@
...
                 },
+                local_overrides: {
+                    discussion: function(element, DiscussionUtil) {
+                        var courseId = $(element).data('course-id');
+
+                        DiscussionUtil.localUrls.push('user_profile');
+                        DiscussionUtil.force_async = true;
+                        DiscussionUtil.route_prefix = 'courses/' + courseId + '/discussion'
+                    }
+                }
```
Comment (antoviaque) [inline jquery.xblock.js]: Wow, we really shouldn't have any code specific to discussion (or any XBlock really) here, this is a standard library and it breaks the abstraction layer. I imagine this is from another feature and was missed during that code review - can you remove the change from this PR, and put a ticket on top of the backlog to remove the change from the client instance too?

**Comment by e-kolpakov** [comment]
@antoviaque sure I can remove it from here, but this is required to keep discussion working with Apros, so we'll need some way to specify this client-specific overrides.

File: jquery.xblock.js (hunk 1 above)
Comment (e-kolpakov) [inline jquery.xblock.js]: @antoviaque sure I can remove it from here, but this is required to keep discussion working with Apros, so we'll need some way to specify this client-specific overrides.

**Comment by mtyaka** [comment]
:+1:

**Pull request merged**